src/server/scripts/Events/hallows_end.cpp

```diff
@@ -0,0 +1,535 @@
+                        {
+                            if (Creature* trigger = me->FindNearestCreature(NPC_FIRE_TRIGGER, 50.0f))
+                                if (!trigger->GetAura(SPELL_START_FIRE) && !trigger->GetAura(SPELL_SPREAD_FIRE))
+                                    DoCast(trigger, SPELL_START_FIRE);
+                            _events.Repeat(Seconds(5));
+                        }
+                        break;
+                    }
+                    case EVENT_STOP_APPLYING_FIRE:
+                        _canShootFire = false;
+                        break;
+                    case EVENT_FINISH:
+                    {
+                        bool failed = false;
+                        std::list<Creature*> triggerList;
+                        me->GetCreatureListWithEntryInGrid(triggerList, NPC_FIRE_TRIGGER);
+                        for (Creature* trigger : triggerList)
+                        {
+                            if (trigger->HasAura(SPELL_FIRE_AURA_BASE))
+                                failed = true;
+
+                            trigger->AI()->EnterEvadeMode();
+                        }
+                        FinishEvent(failed);
+                        break;
+                    }
+                    case EVENT_CLEAVE:
+                        DoCastVictim(SPELL_CLEAVE);
+                        _events.Repeat(Seconds(20));
+                        break;
+                }
+
+            }
+
+            if (!UpdateVictim())
+                return;
+
+            DoMeleeAttackIfReady();
+        }
+
+        void FinishEvent(bool failed)
+        {
+            _events.Reset();
+            if (failed)
+            {
+                Talk(SAY_FAIL);
+                me->DespawnOrUnsummon();
+            }
+            else
+            {
+                Talk(SAY_FLAMES_DOWN);
+                float x, y, z;
+                GetPosToLand(x, y, z);
+                me->GetMotionMaster()->MovePoint(POINT_HORSEMAN_ATTACK, x, y, z);
+            }
+        }
+
+        void MovementInform(uint32 type, uint32 point) override
+        {
+            if (type == POINT_MOTION_TYPE && point == POINT_HORSEMAN_ATTACK)
+            {
+                me->SetFlag(UNIT_FIELD_FLAGS, 0);
+                me->SetDisableGravity(false);
+                me->RemoveAurasDueToSpell(SPELL_HORSEMAN_MOUNT);
+                if (Unit* target = me->SelectNearestPlayer(30.0f))
+                    AttackStart(target);
+            }
+        }
+
+        void JustDied(Unit* /*killer*/) override
+        {
+            Talk(SAY_DEFEATED);
+            float x, y, z;
+            GetPosToLand(x, y, z);
+            me->CastSpell(x, y, z, SPELL_SUMMON_LANTERN, true);
+            CompleteQuest();
+        }
+
+        void CompleteQuest()
+        {
+            float radius = 100.0f;
+            std::list<Player*> players;
+            Trinity::AnyPlayerInObjectRangeCheck checker(me, radius);
```

Comment:
```
src/server/scripts/Events/hallows_end.cpp:395:49:
 fatal error: expected ';' after expression

            Trinity::AnyPlayerInObjectRangeCheck checker(me, radius);
                                                ^
                                                ;
1 error generated.
```
That can not be a correct interpretation by travis-ci - I guess something else needs to be fixed?
From what I can find in search results, that code is used elsewhere in the code without triggering this error.

Reply:
Oh, I see. AppVeyor has got a more detailed explanation.
```
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(395):
 error C2039: 'AnyPlayerInObjectRangeCheck': is not a member of 'Trinity' [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
  C:\projects\trinitycore\src\server\game\Spells\Auras\SpellAuraEffects.h(322):
  note: see declaration of 'Trinity'

  C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(395): error C2065:
 'AnyPlayerInObjectRangeCheck': undeclared identifier [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]

C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(395): error C2146:
 syntax error: missing ';' before identifier 'checker' [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]

C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(395): error C3861:
 'checker': identifier not found [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]

 C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(396): error C2039: 'PlayerListSearcher': is not a member of 'Trinity' [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
  C:\projects\trinitycore\src\server\game\Spells\Auras\SpellAuraEffects.h(322): note: see declaration of 'Trinity'
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(396): error C2065: 'PlayerListSearcher': undeclared identifier [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(396): error C2039: 'AnyPlayerInObjectRangeCheck': is not a member of 'Trinity' [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
  C:\projects\trinitycore\src\server\game\Spells\Auras\SpellAuraEffects.h(322): note: see declaration of 'Trinity'
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(396): error C2065: 'AnyPlayerInObjectRangeCheck': undeclared identifier [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(396): error C2065: 'checker': undeclared identifier [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(396): error C3861: 'searcher': identifier not found [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]

C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(397): error C2039: 'VisitNearbyWorldObject': is not a member of 'Creature' [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
  C:\projects\trinitycore\src\server\game\Spells\SpellScript.h(31): note: see declaration of 'Creature'
C:\projects\trinitycore\src\server\scripts\Events\hallows_end.cpp(397): error C2065: 'searcher': undeclared identifier [C:\projects\trinitycore\build\src\server\scripts\scripts_events.vcxproj]
```

Reply:
OK, so maybe this needs another include, like
```cpp
#include "GridNotifiers.h"
```